File: sha256.java
import java.security.MessageDigest;

class sha256
{

    public static void main(String args[]) 
        throws Exception
    {
        MessageDigest md = MessageDigest.getInstance("SHA-256");
        String text = "Enter values" + " here";
        md.update(text.getBytes());
        byte[] byteData = md.digest();

        StringBuffer sb = new StringBuffer();
        for (int i = 0; i < byteData.length; i++) {
         sb.append(Integer.toString((byteData[i] & 0xff) + 0x100, 16).substring(1));
        }
 
        System.out.println(sb.toString());
    }
}

File: sha256.cpp
#include <stdio.h>
#include <string.h>
#include <openssl/sha.h>

int main()
{
    unsigned char digest[SHA256_DIGEST_LENGTH];
    char string[] = "Enter values here";
    
    SHA256((unsigned char*)&string, strlen(string), (unsigned char*)&digest);    

    char mdString[SHA256_DIGEST_LENGTH*2+1];

    for(int i = 0; i < SHA256_DIGEST_LENGTH; i++)
         sprintf(&mdString[i*2], "%02x", (unsigned int)digest[i]);

    printf("SHA256 digest: %s\n", mdString);

    return 0;
}
